lib/bin/check-file-headers.js

```diff
@@ -40,19 +42,32 @@ process.stdin.on('end', () => {
       throw new Error(`Invalid skip entry: '${skipEntry}'`);
     }));
 
+  const mode = process.argv[2] === '--fix' ? 'fix' : 'check';
+
   let missing = false;
 
+  const hasHeader = new RegExp('^' + header.replace('{{year}}', '20\\d\\d'));
+
   for (const entryPath of files) {
     const contents = fs.readFileSync(entryPath).toString();
-    const withConsistentYear = contents.replace(/(Copyright 20)\d\d/, '$117');
-    if (!withConsistentYear.startsWith(header)) {
+    if (!hasHeader.test(contents)) {
```

Comment:
When I first wrote the script, I thought about using a regular expression for the file header. But if we took that approach, I think we'd want to escape the regular-expression special characters in `header`. Ultimately that felt more complicated to me than replacing each year with 2017.